ratis-server/src/main/java/org/apache/ratis/server/leader/LogAppenderDefault.java

```diff
@@ -73,9 +74,11 @@ private AppendEntriesReplyProto sendAppendEntriesWithRetries()
         }
 
         resetHeartbeatTrigger();
+        final Timestamp sendTime = Timestamp.currentTime();
         getFollower().updateLastRpcSendTime(request.getEntriesCount() == 0);
         final AppendEntriesReplyProto r = getServerRpc().appendEntries(request);
         getFollower().updateLastRpcResponseTime();
+        getFollower().updateLastRespondedAppendEntriesSendTime(sendTime);
```

Comment:
Why still update `LastRespondedAppendEntriesSendTime` at the time of sending?

Reply:
`getServerRpc().appendEntries(request)` is a blocking operation, and once this call returns, the response for the current AppendEntries request has been received. Therefore, we can update its(`LastRespondedAppendEntries`)sendTime.

Reply:
got it~